recipes-core/iotedge-cli/iotedge-cli.inc

```diff
@@ -7,6 +7,6 @@ export LIBIOTHSM_NOBUILD="On"
 do_install () {
     # Binaries
     install -d  "${D}${bindir}"
-    install -m 755 "${WORKDIR}/iotedge-${PV}/edgelet/target/${CARGO_TARGET_SUBDIR}/iotedge" ${D}${bindir}/iotedge
+    install -m 755 "${WORKDIR}/iotedge-${PV}/target/${CARGO_TARGET_SUBDIR}/iotedge" ${D}${bindir}/iotedge
```

Comment:
I am missing some context. What is actually building iotedge-cli?

Reply:
The recipe defines the "iotedge-cli" target. If you look at the [build script](https://github.com/Azure/meta-iotedge/blob/master/scripts/build.sh#L7) , this defines which targets to build.  Normally, the customer would add the layers and include them in their build configuration.  The build script just sets the targets on the command line.

Reply:
It looks like bitbake is the one doing the building of the artifacts based on the targets we specify here. 
https://github.com/Azure/meta-iotedge/blob/master/scripts/build.sh#L20

How does bitbake know how to build the target?

Reply:
There's a lot hidden here. 
1) The layers are installed in the fetch script, and so bitbake will know about the targets.
2) We're dependent on the `meta-rust` layer, which sets up the Rust toolchain and does the actual "cargo build" command.  We start off the recipe with `inherit cargo` and we point to the source directory we want to build by setting `CARGO_SRC_DIR` and the build system takes care of the rest.